### PR TITLE
Feature/add markets functionality

### PR DESCRIPTION
**Pull request by rbUUbr**
1. Added features functionality
2. Enabled Travis CI
3. Added Minitests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
 rvm:
-  - 2.4.1
-before_install: gem install bundler -v 2.0.1
+  - 2.3.1
+  - 2.4.3
+  - 2.5.0
+  - 2.6.2
+notifications:
+  email:
+    - kirabsuir@gmail.com
```
Comment (AlexKlim) [inline .travis.yml]: let's  use craft email

File: test/helpers/example_data.csv [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Asia,Kazakhstan ,Astana International Financial Centre,AIXK,2018-11-18
```
Comment (AlexKlim) [inline test/helpers/example_data.csv]: data file (data/stock_markets.csv) has header and the example doesn't; It seems the file should be with header too

File: README.md [hunk 3]
```diff
@@ -22,7 +21,22 @@ Or install it yourself as:
 
 ## Usage
 
-TODO: Write usage instructions here
+To interact with markets data use class `StockMarkets::Markets`. This class provides `Enumerable` on class level.
+Examples of using:
+
+```ruby
+# To get information about markets as hash, where key - MIC or name of market, value - market data
+StockMarkets::Markets.table #=>  {"AIXK"=>{:continent=>"Asia", :country=>"Kazakhstan ", :name=>"....} }
```
Comment (AlexKlim) [inline README.md:29]: can we have the list/table by stock name too?  (not only by MIC)

**Review by AlexKlim — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (rbUUbr) [reply README.md:29]: We don't have MIC for all stock markets, I think, we can provide additional methods to select stock markets only with mics. WDYT about it? @AlexKlim

Comment (AlexKlim) [reply README.md:29]: yes, you can create a new method to provide stock markets with MIC. 
The idea, to have the ability to find a stock merket by MIC

File: lib/stock_markets/file_data_processor.rb [hunk 4]
```diff
@@ -1,32 +1,61 @@
...
     def initialize
-      @format = 'json'
       @data = []
+      @parsed_csv = nil
     end
 
-    def load_from_disk
-      self.data = CSV.table(File.expand_path("../../data/stock_markets.csv", __FILE__))
+    def load_from_disk!
+      self.parsed_csv = CSV.table(StockMarkets.configuration.data_file_path)
```
Comment (AlexKlim) [inline lib/stock_markets/file_data_processor.rb:18]: what happens if `self.parsed_csv` will not be assigned to data for any reason. looks like the var will be `nil` and we get an exception into `transform_to_hash` method.

File: lib/stock_markets/markets.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    extend Enumerable
+
+    class << self
+      def each(&block)
+        stock_markets_with_mics.values.each do |stock|
+          block.call(stock)
+        end
+      end
+
+      def [](stock_mic)
```
Comment (AlexKlim) [inline lib/stock_markets/markets.rb:12]: why did you remove lookup by `mic_or_name`?

**Review by AlexKlim — CHANGES_REQUESTED** [review]

File: lib/stock_markets/file_data_processor.rb [hunk 6]
```diff
@@ -1,32 +1,61 @@
...
+    def load_data_for_names!
+      transform_to_hash(key_name: MARKET_NAME_CSV_COLUMN).data
+    end
+
+    private
+
+    attr_accessor :parsed_csv
+
+    def last_update_market_minimal_date
+      return @_last_update_market_minimal_date if @_last_update_market_minimal_date
```
Comment (AlexKlim) [inline lib/stock_markets/file_data_processor.rb:45]: dont you like 

```ruby
@_last_update_market_minimal_date ||= begin 
  ....
end
```
? :)

Comment (rbUUbr) [reply lib/stock_markets/file_data_processor.rb:45]: @AlexKlim https://github.com/rubocop-hq/ruby-style-guide#no-BEGIN-blocks I think, it is not good idea :(

File: lib/stock_markets/markets.rb (hunk 5 above)
Comment (rbUUbr) [reply lib/stock_markets/markets.rb:12]: Discussed. Resolved

File: lib/stock_markets/file_data_processor.rb (hunk 4 above)
Comment (rbUUbr) [reply lib/stock_markets/file_data_processor.rb:18]: Great catch. I will move it to initializer

File: lib/stock_markets/file_data_processor.rb (hunk 6 above)
Comment (AlexKlim) [reply lib/stock_markets/file_data_processor.rb:45]: I need more details :)

Comment (rbUUbr) [reply lib/stock_markets/file_data_processor.rb:45]: just avoid of using this :D

**Review by AlexKlim — APPROVED** [review]
